tests/modules/programs/wayland-bongocat/custom-settings.nix

```diff
@@ -0,0 +1,41 @@
+{ config, pkgs, ... }:
+{
+  config = {
+    programs.wayland-bongocat = {
+      package = config.lib.test.mkStubPackage { name = "wayland-bongocat"; };
+      enable = true;
+      cat = {
+        height = 123;
+      };
+      keyboardDevices = [
+        "/dev/input/event1"
+        "/dev/input/event2"
+      ];
+    };
+
+    nmt.script =
+      let
+        configDir =
+          if pkgs.stdenv.isDarwin then
+            "home-files/Library/Application Support/wayland-bongocat"
+          else
+            "home-files/.config/wayland-bongocat";
+        configFile = "${configDir}/settings.conf";
+      in
+      ''
+        assertFileExists "${configFile}"
+        assertFileContent "${configFile}" ${pkgs.writeText "wayland-bongocat.config-custom.expected" ''
+          # Generated by Home Manager.
+          [main]
+          cat_height=123
+          cat_x_offset=0
+          cat_y_offset=0
+          enable_debug=0
+          fps=60
+          overlay_opacity=150
+
+          keyboard_device=/dev/input/event1
+          keyboard_device=/dev/input/event2''}
```

Comment:
I'm not done yet, but if somebody sees this, I'm not happy with the empty line and the missing newline at the end.